packages/transactions/tests/builder.test.ts

```diff
@@ -2359,3 +2370,279 @@ test.each([
 
   expect(bytesToHex(transaction.serialize())).toEqual(txBytes);
 });
+
+describe('multi-sig', () => {
+  test('working legacy (sequential) multi-sig', async () => {
+    const pk1 = createStacksPrivateKey(bytesToHex(randomBytes(32)) + '01');
+    const pk2 = createStacksPrivateKey(bytesToHex(randomBytes(32)) + '01');
+    const pk3 = createStacksPrivateKey(bytesToHex(randomBytes(32)) + '01');
+
+    const publicKeys = [pk1, pk2, pk3].map(pk => publicKeyToString(getPublicKey(pk)));
+    const signerKeys = [pk1, pk2].map(pk => privateKeyToString(pk));
+
+    const tx = await makeSTXTokenTransfer({
+      recipient: 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6',
+      amount: 12345n,
+      fee: 1_000_000n,
+      nonce: 0n,
+      network: 'testnet',
+      anchorMode: AnchorMode.Any,
+      numSignatures: 2,
+      publicKeys,
+      signerKeys,
+    });
+
+    expect(() => tx.verifyOrigin()).not.toThrow();
```

Comment:
In addition to `verifyOrigin()` you could add a few asserts at the end of these tests, like check that we have the expected number of auth fields and signatures, that they match the keys we passed, etc. Right now `MakeSTXTokenTransfer()` could be ignoring most of its args and returning a valid singlesig tx, and these unit tests would still pass